### DISCUSSION TITLE
Flush all automationNode ?

### DISCUSSION THREAD

**Discussion by Tenjim** [AvaloniaUI/Avalonia#16401] [Q&A]
Hi guys,

I have a question about AutomationNode.
I use Windows UIAutomation and DotMemoryUnit to verify leaks on our models and I saw some retention on avalonia's AutomationNode. In the past, In wpf, I detached, called the GC, the reattached the process to flush all automationNode / AutomationPeer like [this issue](https://stackoverflow.com/questions/50641406/memory-leak-with-custom-frameworkelementautomationpeer-with-iexpandcollapseprovi) 

However in Avalonia, this workaround has no effect, and so AutomationPeer objects are still there with strong reference on my model :

![image](https://github.com/user-attachments/assets/cff6f699-ea5a-47b0-b9ee-05a838f7d54f)

There is a way to flush all AutomationNode/AutomationPeer objects when we detach this process ?

**Comment by maxkatz6** [comment]
I don't think WPF hacks can be applied here.
But there seems to be a leak with automation somewhere, which is tracked in https://github.com/AvaloniaUI/Avalonia/issues/8240 issue
